src/Tree/Tree.js

```diff
@@ -1,238 +1,330 @@
-import { Dropdown } from '../Dropdown/Dropdown';
+import classnames from 'classnames';
 import PropTypes from 'prop-types';
+import shortid from '../utils/shortId';
 import React, { Component } from 'react';
 
-export class Tree extends Component {
+export class TreeCell extends Component {
+    render() {
+        const {
+            children,
+            className,
+            ...rest
+        } = this.props;
+
+        const cellClassName = classnames(
+            'fd-tree__col',
+            className
+        );
+
+        return (
+            <div
+                {...rest}
+                className={cellClassName}>
+                {children}
+            </div>
+        );
+    }
+}
+
+TreeCell.propTypes = {
+    children: PropTypes.node,
+    className: PropTypes.string
+};
+
+export class TreeHeader extends Component {
+    render() {
+        const {
+            buttonProps,
+            children,
+            isExpanded,
+            onExpandAll,
+            ...rest
+        } = this.props;
+
+        return (
+            <div {...rest} className='fd-tree fd-tree--header'>
+                <div className='fd-tree__row fd-tree__row--header'>
+                    {
+                        React.Children.map(children, (child, index) => {
+                            const isFirstTreeCell = index === 0 && child.type && child.type.name === 'TreeCell';
+
+                            // Add control class to first TreeCell element
+                            const className = classnames({
+                                'fd-tree__col--control': isFirstTreeCell
+                            });
+
+                            // Add expand button to first TreeCell element
+                            const newChildren = isFirstTreeCell ? (
+                                <div>
+                                    <button
+                                        {...buttonProps}
+                                        aria-label='expand'
```

Comment:
This should be extracted into a localized prop although that's probably a whole other story in itself since none of the components support that right now.  Also, the label should probably also toggle between "expand" and "collapse".

Reply:
I will leave it hard coded for now so we can tackle those localized labels at once. Toggling the labels to "expand"/"collapse" in a2bd47e